GiftModels/FinancialDetails.cs

```diff
@@ -40,12 +40,13 @@ public class FinancialDetails
                 string retval = null;
                 if (string.Equals(FundType, "CURRENT", StringComparison.OrdinalIgnoreCase) && !(string.Equals(Purpose, "LOAN", StringComparison.OrdinalIgnoreCase) || string.Equals(Purpose, "CAPITAL", StringComparison.OrdinalIgnoreCase)))
                 {
-                    retval =  ExpenseChart + "-" + ExpenseAccount;
+                    if (!string.IsNullOrWhiteSpace(ExpenseChart) && !string.IsNullOrWhiteSpace(ExpenseChart))

```

Comment:
@taylorkj You need to check ExpenseAccount. You checked the chart twice.
Also, will this ever have an account without a chart?